pilot/pkg/config/file/store.go

```diff
@@ -408,6 +412,32 @@ func (s *KubeSource) parseChunk(r *collection.Schemas, name string, lineNum int,
 	}, nil
 }
 
+var allowedVersion map[string]bool
+
+func findSchemaByGVK(r *collection.Schemas, chunkSchema *schema.GroupVersionKind,
+	allowVersionConversion bool) (collection.Schema, bool) {
+	if allowedVersion == nil {
+		allowedVersion = map[string]bool{}
+		for _, s := range collections.All.All() {
+			allowedVersion[s.Resource().Version()] = true
+		}
+	}
+	sc, found := r.FindByGroupVersionKind(schemaresource.FromKubernetesGVK(chunkSchema))
+	if !found && allowVersionConversion {
+		gvk := &schema.GroupVersionKind{
+			Group: chunkSchema.Group,
+			Kind:  chunkSchema.Kind,
+		}
+		for version := range allowedVersion {
+			gvk.Version = version
+			if sc, found = r.FindByGroupVersionKind(schemaresource.FromKubernetesGVK(gvk)); found {
```

Comment:
Looks like we are doing this for ANY type. But its only safe for some types -- really the `istio.io` ones. Maybe in metadata.yaml we can can put some field for version aliases, and for Istio types add them.

For other types, the versions are not inter-changable.

Reply:
Yep, I thought there might be configuration changes between packages with different versions. Thanks for the point, I'll look into it.